chat-plugins/scavengers.js

```diff
@@ -783,8 +783,27 @@ let commands = {
 		if (!room.game || !room.game.scavGame) return this.errorReply(`There is no scavenger game currently running.`);
 
 		let game = room.game.childGame || room.game;
-		let elapsed = Date.now() - game.startTime;
-		this.sendReplyBox(`The current ${game.gameType ? `<em>${game.gameType}</em> ` : ""}scavenger hunt by <em>${Chat.escapeHTML(Chat.toListString(game.hosts.map(h => h.name)))}${(game.hosts.some(h => h.userid === game.staffHostId) ? '' : ` (started by - ${Chat.escapeHTML(game.staffHostName)})`)}</em> has been up for: ${Chat.toDurationString(elapsed, {hhmmss: true})}<br />Completed (${game.completed.length}): ${game.completed.map(u => Chat.escapeHTML(u.name)).join(', ')}<br />`);
+		if (game.hosts.some(h => h.userid === user.userid) || game.staffHostId === user.userid) {
```

Comment:
Like why is `game.hosts` retaining references to users? That should not ever happen.

Reply:
You could just keep userid's instead of actual players for everything, but then we run into problems with people joining hunts that they created on their own alts and finishing them.

Reply:
People shouldn't be changing accounts in the middle of a game, it confuses users just as much.

Reply:
Changing names means considering them different users, that's the point of changing names!

Reply:
https://github.com/Zarel/Pokemon-Showdown/blob/master/chat-plugins/scavengers.js#L540
each host object only consists of the a "cloned" name and userid

Reply:
Oh, okay.